homeassistant/components/romy/__init__.py

```diff
@@ -0,0 +1,38 @@
+"""ROMY Integration."""
+
+import romy
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, PLATFORMS
+from .coordinator import RomyVacuumCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Initialize the ROMY platform via config entry."""
+
+    password = config_entry.data.get(CONF_PASSWORD, "")
+
+    new_romy = await romy.create_romy(config_entry.data[CONF_HOST], password)
+
+    name = config_entry.data[CONF_NAME]
+    if name != new_romy.name:
+        await new_romy.set_name(name)
```

Comment:
I thought I left a comment about this but I can't find it now so maybe I didn't.

I think all aspects of asking for a name in the config flow and setting a device name here should be removed -- instead the name should be retrieved from the device. Then, the user can use built in UI in home assistant for changing the name in home assistant only. (We never want home assistant to clobber the devices name if it changed).

I would suggest removing this as well as asking for name in the config flow.

Reply:
maybe you were thinking about the comment with attr names (self._device_info[ATTR_NAME] = self.romy.name)

removed name setup in flow control + adjust zero conf so it only asks confirmation for adding if robot is already unlocked

zeroconf setup:
![new_zeroconf](https://github.com/home-assistant/core/assets/2132412/934cc47c-6791-4155-b04c-53d3bcd34029)

userconf setup:
![new_userconf](https://github.com/home-assistant/core/assets/2132412/114b60c8-a850-459f-be10-4f0ef505da72)